contracts/Manager.sol

```diff
@@ -617,4 +672,43 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
             }
         }
     }
+
+    /**
+     * @notice Marks a group as deprecated.
+     * @param group The group to deprecate.
+     */
+    function _deprecateGroup(address group) private {
+        if (!activeGroups.remove(group)) {
+            revert GroupNotActive(group);
+        }
+
+        emit GroupDeprecated(group);
+
+        if (account.getCeloForGroup(group) > 0) {
+            if (!deprecatedGroups.add(group)) {
+                revert FailedToAddDeprecatedGroup(group);
+            }
+        } else {
+            emit GroupRemoved(group);
+        }
+    }
+
+    /**
+     * @notice Checks if a group member is elected.
+     * @param groupMember The member of the group to check election status for.
+     * @return Whether or not the group member is elected.
+     */
+    function isGroupMemberElected(address groupMember) private view returns (bool) {
+        IElection election = getElection();
+
+        address[] memory electedValidatorSigners = election.electValidatorSigners();
+
+        for (uint256 i = 0; i < electedValidatorSigners.length; i++) {
+            if (electedValidatorSigners[i] == groupMember) {
+                return true;
+            }
+        }
```

Comment:
I guess I'm still a bit worried about the gas costs of this check. We are reading O(100) values from storage, then potentially iterating over them all.

Additionally, since we're simulating an election, we're actually answering the question "*would* this validator be elected if we ran an election *right now*", not if they're actually currently elected. In practice, the elected set doesn't change often and this is likely to give us the right answer. I could see a minor and hard to pull off potential attack:
1. Eve gets validators from two groups elected.
2. Later, one of the groups loses a lot of votes, but Eve herself has enough votes to keep one group elected.
3. When someone tries to deprecate the group that doesn't have enough votes, she frontruns the transaction, changing her votes to the other group.

Definitely an unlikely attack vector (requires front running, doesn't really give a large benefit since in this scenario Eve doesn't gain an extra elected group), but something to be aware of.